integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
+		if op.Name == "loop" {
+			return true
+		}
+	}
+	return false
+}
+
+// marshalStructToFile marshals the given object and writes to filePath
+func marshalStructToFile(t *testing.T, obj interface{}, filePath string) {
+	t.Helper()
+	// MarshalExtJSON is used to print out bson.Raw properly
+	marshaled, err := bson.MarshalExtJSON(obj, false, false)
+	if err != nil {
+		t.Fatalf("marshal results failed: %v", err)
+	}
+	err = ioutil.WriteFile(filePath, marshaled, 0644)
+	if err != nil {
+		t.Fatalf("write to file failed: %v", err)
+	}
+}
+
+func TestAtlasPlannedMaintenance(t *testing.T) {
+	connstring := os.Args[1]
+	workloadSpec := []byte(os.Args[2])
+
+	setupOpts := mtest.NewSetupOptions().SetURI(connstring)
+	if err := mtest.Setup(setupOpts); err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if err := mtest.Teardown(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// killAllSessions will return an auth error if it's run
+	fileReqs, testCases := unified.ParseTestFile(t, workloadSpec, unified.NewOptions().SetRunKillAllSessions(false))
```

Comment:
While this does work, please note that https://jira.mongodb.org/browse/DRIVERS-1577 requires the unified test runner to ignore permission errors on killing sessions.

Reply:
removed. This was written before the go ticket for that was done.